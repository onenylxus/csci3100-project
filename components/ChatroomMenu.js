// Import
import React from 'react';
import { View, Text, TouchableOpacity, Image, Dimensions } from 'react-native';
import { Grid, Col } from 'react-native-easy-grid';
import { useNavigation } from '@react-navigation/native';
import Style from '../assets/style';

// export individual chatBox in chatroom Menu
export default function Chat() {
  const navigation = useNavigation();
  const windowWidth = Dimensions.get('window').width;

  function chatNavigation(screenWidth) {
    if (screenWidth < 1100) {
      return () => navigation.navigate('ChatboxPhone');
    }
    return null;
  }

  return (
    <View style={Style.chatRoomContainerPhone}>
      <TouchableOpacity
        style={Style.chatBox}
        onPress={chatNavigation(windowWidth)} // if using small screen, navigate to Chat
      >
        <Grid>
          <Col style={{ alignSelf: 'center' }} size={1}>
            <Image
              style={Style.chatMenuIcon}
              source={require('../assets/images/defaultProfile.png')}
            />
          </Col>
          <Col style={{ alignSelf: 'center' }} size={2}>
            <View style={Style.chatMenuInfo}>
              <Text style={{ fontSize: 18, marginBottom: 2 }}>Username</Text>
              <Text style={{ fontSize: 13 }}>Recent message</Text>
            </View>
          </Col>
          <Col style={{ alignSelf: 'center' }}>
            <View>
              <Text style={{ alignSelf: 'flex-end', marginRight: 15 }}>
                12:30
              </Text>
            </View>
          </Col>
        </Grid>
      </TouchableOpacity>
    </View>
  );
}
