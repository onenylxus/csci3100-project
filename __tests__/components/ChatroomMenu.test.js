// Import
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render } from '@testing-library/react-native';

// Import target component
import ChatroomMenu from '../../components/ChatroomMenu';

// Mock navigation route
jest.mock('@react-navigation/core');

// Run jest tests
describe('ChatroomMenu', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(
      <NavigationContainer>
        <ChatroomMenu />
      </NavigationContainer>
    );
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
