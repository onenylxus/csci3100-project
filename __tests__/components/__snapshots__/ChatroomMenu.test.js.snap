// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatroomMenu matches snapshot 1`] = `null`;
